### PR TITLE
Update product component on button click test

### PR DESCRIPTION
**Pull request by jameskim08**
Updated onButtonClick in product.js
- Created tests for when buttonDestination is a function or modal
- Separated existing test for cart buttonDestination into smaller tests
- Added tests for function calls
- A test for when config.cart.popup is false was not made because testing the window location is not possible

### CODE REVIEW

File: test/unit/product/product-component.js [hunk 1]
```diff
@@ -401,66 +410,174 @@ describe('Product Component class', () => {
...
-            });
+        it('calls buttonDestination if it is a function', () => {
+          const buttonDestinationSpy = sinon.spy();
+          product.config.product.buttonDestination = buttonDestinationSpy;
+          product.onButtonClick(evt, target);
+          assert.calledOnce(buttonDestinationSpy);
+          assert.calledWith(buttonDestinationSpy, product);
+        });
+
+        describe('if button destination is cart', () => {
```
Comment (spencercanner) [inline test/unit/product/product-component.js]: Are we missing a test for closing the modal?

File: test/unit/product/product-component.js [hunk 2]
```diff
@@ -401,66 +410,174 @@ describe('Product Component class', () => {
...
+            assert.calledWith(setActiveElSpy, target);
+          });
+        });
+
+        describe('if button destination is modal', () => {
+          let openModalStub;
+
+          beforeEach(() => {
+            product.config.product.buttonDestination = 'modal';
+            openModalStub = sinon.stub(product, 'openModal');
```
Comment (spencercanner) [inline test/unit/product/product-component.js]: Does this need to be restored?

File: test/unit/product/product-component.js [hunk 3]
```diff
@@ -401,66 +410,174 @@ describe('Product Component class', () => {
...
+          it('opens checkout in a new window if cart popup in config is true', async () => {
+            product.config.cart.popup = true;
+            const checkout = new Checkout(product.config);
+            product.onButtonClick(evt, target);
+            await Promise.all([createCheckoutPromise, addLineItemsPromise]);
+            assert.calledOnce(openWindowStub);
+            assert.calledWith(openWindowStub, '', 'checkout', checkout.params);
+          });
+
+          it('create checkout and add line items are called when destination is checkout', async () => {
```
Comment (spencercanner) [inline test/unit/product/product-component.js]: `it('creates ... `

File: test/unit/product/product-component.js [hunk 4]
```diff
@@ -401,66 +410,183 @@ describe('Product Component class', () => {
...
+
+          it('opens checkout in a new window if cart popup in config is true', async () => {
+            product.config.cart.popup = true;
+            const checkout = new Checkout(product.config);
+            product.onButtonClick(evt, target);
+            await Promise.all([createCheckoutPromise, addLineItemsPromise]);
+            assert.calledOnce(openWindowStub);
+            assert.calledWith(openWindowStub, '', 'checkout', checkout.params);
+          });
+
```
Comment (melissaluu) [inline test/unit/product/product-component.js]: Are we missing a test that covers when checkout isn't a popup?

File: test/unit/product/product-component.js [hunk 5]
```diff
@@ -401,66 +410,183 @@ describe('Product Component class', () => {
...
+            product.onButtonClick(evt, target);
+            await Promise.all([createCheckoutPromise, addLineItemsPromise]);
+
+            assert.calledOnce(createCheckout);
+            assert.calledOnce(addLineItems);
+            assert.calledWith(addLineItems, checkoutMock.id, [{
+              variantId: 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0VmFyaWFudC8xMjM0NQ==',
+              quantity: selectedQuantity,
+            }]);
+          });
```
Comment (melissaluu) [inline test/unit/product/product-component.js]: Can we check if the window has the right location?

File: test/unit/product/product-component.js [hunk 6]
```diff
@@ -401,66 +410,183 @@ describe('Product Component class', () => {
...
+          product.config.product.buttonDestination = 'onlineStore';
+          product.onButtonClick(evt, target);
+          assert.calledOnce(openOnlineStoreStub);
+          openOnlineStoreStub.restore();
+        });
+
+        describe('if button destination is checkout', () => {
+          let createCheckout;
+          let createCheckoutPromise;
           let addLineItems;
```
Comment (melissaluu) [inline test/unit/product/product-component.js]: Let's be consistent and keep the variable naming consistent when it comes to spies and stubs.

**Pull request merged**